pootle/apps/pootle_statistics/utils.py

```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) Pootle contributors.
+#
+# This file is a part of the Pootle project. It is distributed under the GPL3
+# or later license. See the LICENSE file for a copy of the license and the
+# AUTHORS file for copyright and authorship information.
+
+from collections import OrderedDict
+
+from django.contrib.auth import get_user_model
+from django.db.models import Count, Q
+from django.utils.functional import cached_property
+
+
+User = get_user_model()
+
+
+class Contributors(object):
+
+    def __init__(self, include_anon=False,
+                 project_codes=None, language_codes=None,
+                 since=None, until=None, sort_by="username"):
+        self.include_anon = include_anon
+        self.project_codes = project_codes
+        self.language_codes = language_codes
+        self.since = since
+        self.until = until
+        self.sort_by = sort_by
+
+    @property
+    def user_qs(self):
+        if self.include_anon:
+            return User.objects.exclude(username__in=["system", "default"])
```

Comment:
no - this is relevant to this class - unless we have a need for this pattern elsewhere im going to keep it here for now